src/couch_index/src/couch_index_compactor.erl

```diff
@@ -81,6 +85,14 @@ handle_cast(_Mesg, State) ->
 
 handle_info({'EXIT', Pid, normal}, #st{pid=Pid}=State) ->
     {noreply, State#st{pid=undefined}};
+handle_info({'EXIT', Pid, Reason}, #st{pid = Pid} = State) ->
+    #st{idx = Idx, mod = Mod} = State,
+    {ok, IdxState} = gen_server:call(Idx, {compaction_failed, Reason}),
```

Comment:
we don't do that with gen_call for [`compacted`](https://github.com/cloudant/couchdb/blob/a39bfd1d0529fda9415229ff2b0698837988f332/src/couch_index/src/couch_index_compactor.erl#L126) though. I think the intent here is _not_ to expose internals of `couch_index` compactor's flow, we want public API on triggering compaction, but not on interrupting its execution except from compaction process.